<?xml version="1.0" ?>
<launch>
    <!-- include file="$(find agv_low)/launch/agv_low_youbot_collisionwheel.launch" /-->
    <node name="controller" pkg="pid" type="controller" ns="/agv_mecanum/pid_x" output="screen" >
      <param name="Kp" value="3.0" />
      <param name="Ki" value="1.35" />
      <param name="Kd" value="0.032" />
      <param name="upper_limit" value="0.5" />
      <param name="lower_limit" value="-0.5" />
      <param name="windup_limit" value="0.01" />
      <param name="cutoff_frequency" value="3" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="50.0" />
      <param name="min_loop_frequency" value="15.0" />
     </node>
    <node name="controller" pkg="pid" type="controller" ns="/agv_mecanum/pid_y" output="screen" >
      <param name="Kp" value="1.0" />
      <param name="Ki" value="0.7" />
      <param name="Kd" value="0.09" />
      <param name="upper_limit" value="0.5" />
      <param name="lower_limit" value="-0.5" />
      <param name="windup_limit" value="0.01" />
      <param name="cutoff_frequency" value="3" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="50.0" />
      <param name="min_loop_frequency" value="15.0" />
     </node>
     <node name="controller" pkg="pid" type="controller" ns="/agv_mecanum/pid_yaw" output="screen" >
      <param name="kp" value="0.1" />
      <param name="ki" value="1.35" />
      <param name="kd" value="0.032" />
      <param name="upper_limit" value="0.5" />
      <param name="lower_limit" value="-0.5" />
      <param name="windup_limit" value="0.2" />
      <param name="cutoff_frequency" value="3" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="50.0" />
      <param name="min_loop_frequency" value="15.0" />
     </node>
     <node name="controller" pkg="pid" type="controller" ns="/agv_mecanum/pid_vel" output="screen" >
      <param name="kp" value="1.0" />
      <param name="ki" value="0.0" />
      <param name="kd" value="0.0" />
      <param name="upper_limit" value="0.5" />
      <param name="lower_limit" value="-0.5" />
      <param name="windup_limit" value="0.2" />
      <param name="cutoff_frequency" value="3" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="50.0" />
      <param name="min_loop_frequency" value="15.0" />
     </node>
    <!-- node pkg="position_controller" type="setpoint_broadcaster.py" name="setpoint_broadcaster" output="screen"/-->
    <!-- node pkg="position_controller" type="trajectory_planner.py" name="trajectory_planner" output="screen"/ -->
    <node pkg="position_controller" type="position_controller.py" name="position_controller" output="screen"/>
</launch>
